src/main/java/org/baeldung/security/LoggedUser.java

```diff
@@ -0,0 +1,49 @@
+package org.baeldung.security;
+
+import java.util.List;
+
+import javax.servlet.http.HttpSessionBindingEvent;
+import javax.servlet.http.HttpSessionBindingListener;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.authentication.event.AuthenticationSuccessEvent;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.session.SessionDestroyedEvent;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+@Component
+public class LoggedUser implements HttpSessionBindingListener {
```

Comment:
May be we can add a constructor that takes username as parameter. That we, you will save yourself an extra line every time you create LoggedUser instance.